### ISSUE TITLE
Sub-classed constraint doesn't appear to be called

### ISSUE THREAD

**Issue by EnderWiggin14** [tensorflow/tensorflow#51922]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): A proprietary variant of RedHat, outside of my control
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.5.0
- Python version: 3.7.2
- CUDA/cuDNN version: 11.1.0/8.1.1
- GPU model and memory: Nvidia K40 12GB

**Describe the current behavior**
I have created a sub-classed layer that performs a weighted sum of the outputs of two sub_models.
In this layer, I create weights with the `add_weight` function which includes the use of a constraint.

The intention of the constraint is to normalize the weights of each input dimension/feature (i.e. sum to one).

I have created a custom weight normalization constraint which is sub-classed from `krs.constraints.Constraint`. The output weights from the constraint function for model consisting of 2 sub-models with each producing a 200 dimension/feature output. Thus, the input to the constraint would be 2 x 200 weight tensor and the columnwise sums should all be 1.0.

When providing an instance of my constraint class to the `add_weight` function, it appears that there it is never called. It is initialized though. When running `VectorWeightedSum_1` below the `tf.print` statement in the call function shows that the weights are never normalized. Also, the `tf.print` statement in `NormalizeSumWeights.call()` doesn't ever produce an output. (It also doesn't exit the program if I put a `sys.exit()` call inside the contraint's `call`)

If I instead use `VectorWeightedSum_2`, the `tf.print` statement demonstrates that weights are indeed normalized properly. This was done by creating a new function inside the `VectorWeightedSum_2` class as opposed to a constraint class. The line for this code is commented out in the code below.

**Describe the expected behavior**
The sub-classed constraint should have it's `call` function called.

The weights produced by the sub-classed constraint should sum to 1.0 along each column.

In the example code, the `tf.print` statements should be printed for both `_constraint` and the use of the `NormalizeSumWeights` class.

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): No

**Standalone code to reproduce the issue**
```
import numpy as np
import tensorflow as tf
import tensorflow.keras as krs

class NormalizeSumWeights(krs.constraints.Constraint):
    def __init__(self,**kwargs):
        super(NormalizeSumWeights,self).__init__(**kwargs)

    def call(self,w):
        tf.print("NormalizeSumWeights_________________________________")
        return tf.math.divide(w,tf.math.reduce_sum(w,axis=0))

class WeightedSum(krs.layers.Layer):
    def __init__( self, n_models = 2, **kwargs):
        super( WeightedSum, self ).__init__( **kwargs)
        self.n_models = n_models

    def build(self,input_shape):
        self.sum_weights = self.add_weight(name="sum_weights",shape=(self.n_models,1),
                                    initializer = krs.initializers.RandomUniform(0,1.0),
                                    constraint = NormalizeSumWeights(),
                                    trainable = True)

    def call(self,inputs):
        output = tf.multiply(tf.cast(self.sum_weights[0,:],inputs[0].dtype),inputs[0])
        for i in range(1,len(inputs)):
            output += tf.multiply(inputs[i],tf.cast(self.sum_weights[i,0],inputs[0].dtype))

        return output

    def get_config(self):
        data = { "n_models": self.n_models}
        return data

class VectorWeightedSum_1(krs.layers.Layer):
    def __init__( self, n_models = 2,n_dimensions=200, **kwargs):
        super( VectorWeightedSum_1, self ).__init__( **kwargs)
        self.n_models = n_models
        self.n_dimensions = n_dimensions

    def build(self,input_shape):
        self.sum_weights = self.add_weight(name="sum_weights",shape=(self.n_models,self.n_dimensions),
                                    initializer = krs.initializers.RandomUniform(0,1.0),
                                    constraint = NormalizeSumWeights(),
                                    trainable = True)
    def _constraint(self,w):
        tf.print("Weights Normalization__________________________")
        return tf.math.divide(w,tf.math.reduce_sum(w,axis=0))

    def call(self,inputs):
        output = tf.multiply(tf.cast(self.sum_weights[0,:],inputs[0].dtype),inputs[0])

        for i in range(1,len(inputs)):
            output += tf.multiply(inputs[i],tf.cast(self.sum_weights[i,:],inputs[0].dtype))
        tf.print("vectorWeightedSum___weights________________")
        tf.print(self.sum_weights)

        return output

class VectorWeightedSum_2(krs.layers.Layer):
    def __init__( self, n_models = 2,n_dimensions=200, **kwargs):
        super( VectorWeightedSum_2, self ).__init__( **kwargs)
        self.n_models = n_models
        self.n_dimensions = n_dimensions

    def build(self,input_shape):
        self.sum_weights = self.add_weight(name="sum_weights",shape=(self.n_models,self.n_dimensions),
                                    initializer = krs.initializers.RandomUniform(0,1.0),
                                    constraint = self._constraint,
                                    trainable = True)
    def _constraint(self,w):
        tf.print("Weights Normalization__________________________")
        return tf.math.divide(w,tf.math.reduce_sum(w,axis=0))

    def call(self,inputs):

        output = tf.multiply(tf.cast(self.sum_weights[0,:],inputs[0].dtype),inputs[0])

        for i in range(1,len(inputs)):
            output += tf.multiply(inputs[i],tf.cast(self.sum_weights[i,:],inputs[0].dtype))
        tf.print("vectorWeightedSum___weights________________")
        tf.print(self.sum_weights)

        return output

input_lf = krs.Input((4,))

x = input_lf
x = krs.layers.Dense(10,activation = 'relu')(x)
x = krs.layers.Dense(10,activation = 'relu')(x)
lf_out = krs.layers.Dense(200,activation = 'relu')(x)

lf_mod = krs.Model(input_lf,lf_out,name='lf')

input_hf_lin = krs.Input((204,))
x = input_hf_lin
x = krs.layers.Dense(10)(x)
x = krs.layers.Dense(10)(x)
hf_lin_out = krs.layers.Dense(200,activation = 'relu')(x)

hf_lin_mod = krs.Model(input_hf_lin,hf_lin_out,name='hf_linear')

input_hf_nonlin = krs.Input((14,))

x = input_hf_nonlin
x = krs.layers.Dense(10,activation = 'relu')(x)
x = krs.layers.Dense(10,activation = 'relu')(x)
hf_nonlin_out = krs.layers.Dense(200,activation = 'relu')(x)

hf_nonlin_mod = krs.Model(input_hf_lin,hf_lin_out,name='hf_nonlinear')

input_hf = krs.Input((204,))

x = input_hf
lin = hf_lin_mod(x)
nonlin = hf_nonlin_mod(x)
summed_out = VectorWeightedSum_1(n_models=2)([lin,nonlin])
#summed_out = VectorWeightedSum_2(n_models=2)([lin,nonlin])

hf_mod = krs.Model(input_hf,summed_out,name='hf')

input_full_mod = krs.Input((4,))
x = input_full_mod

low = lf_mod(x)
x = krs.layers.Concatenate()([low,x])
full_out = hf_mod(x)

full_mod = krs.Model(input_full_mod,outputs = {'low_fidelity':low,'high_fidelity':full_out},name='full_model')

opt = krs.optimizers.Adam()
loss = krs.losses.MSE
full_mod.compile(optimizer = opt,loss = loss)

x_train = np.random.uniform(0,10,(20,4))
y_train_low = np.random.uniform(0,10,(20,200))
y_train_high = np.random.uniform(0,10,(20,200))
y = {"low_fidelity": y_train_low,
     "high_fidelity": y_train_high}

full_mod.fit(x_train,y,epochs=5,batch_size=1)
```

**Comment by sushreebarsa** [comment]
@EnderWiggin14 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by EnderWiggin14** [comment]
Bug issue moved [here](https://github.com/keras-team/keras/issues/15345).
